canvas_modules/common-canvas/src/common-properties/util/property-utils.js

```diff
@@ -276,7 +276,7 @@ function convertArrayStructureToObject(isList, subControls, currentValues, conve
 		structureKeys.push(control.name);
 	});
 
-	if (isList) {
+	if (isList && currentValues && Array.isArray(currentValues)) {
```

Comment:
You can remove the `currentValues` check.  `Array.isArray(currentValues)` handles when values are undefined or null.